lib/active_job/jobs_relation.rb

```diff
@@ -51,12 +51,16 @@ def initialize(queue_adapter: ActiveJob::Base.queue_adapter, default_page_size:
   # * <tt>:queue_name</tt> - To only include the jobs in the provided queue.
   # * <tt>:worker_id</tt> - To only include the jobs processed by the provided worker.
   # * <tt>:recurring_task_id</tt> - To only include the jobs corresponding to runs of a recurring task.
-  def where(job_class_name: nil, queue_name: nil, worker_id: nil, recurring_task_id: nil)
+  # * <tt>:finished_at_start</tt> - To only include the jobs finished after the provided date.
+  # * <tt>:finished_at_end</tt> - To only include the jobs finished before the provided date.
+  def where(job_class_name: nil, queue_name: nil, worker_id: nil, recurring_task_id: nil, finished_at_start: nil, finished_at_end: nil)
```

Comment:
I'd represent `finished_at` with a range here in `JobRelation`, instead of two values.

Reply:
I was giving this a try but got a little bit stuck.

I think `finshed_at_start` and `finished_at_end` should be different parameters for `where` clauses concatenation. Also having them as separated properties makes the `clone_with` logic simpler

Maybe, we should encapsultate them in a `finshed_at` filter, eg: make `ActiveJob::QueueAdapters::SolidQueueExt#supported_job_filtetrs` return finished_at instead of both `finshed_at_start` and `finished_at_end`

Reply:
> we should encapsultate them in a finshed_at filter, eg: make ActiveJob::QueueAdapters::SolidQueueExt#supported_job_filtetrs return finished_at instead of both finshed_at_start and finished_at_end

Yes, definitely!

> I think finshed_at_start and finished_at_end should be different parameters for where clauses concatenation. Also having them as separated properties makes the `clone_with` logic simpler

I think you can do this in the same way if you set `finished_at` as [a range](https://docs.ruby-lang.org/en/master/Range.html). It can be a beginless or endless range, and you can still concatenate `where` clauses with that, and the `clone_with` logic shouldn't change.

Reply:
> 
> I think you can do this in the same way if you set `finished_at` as [a range](https://docs.ruby-lang.org/en/master/Range.html). It can be a beginless or endless range, and you can still concatenate `where` clauses with that, and the `clone_with` logic shouldn't change.

would you mind sharing some code implementing? I am not sure how to do it

Reply:
Something like this? It's not completely done and I haven't tested it, but I think something like this should work:
```diff
diff --git a/app/controllers/concerns/mission_control/jobs/job_filters.rb b/app/controllers/concerns/mission_control/jobs/job_filters.rb
index 01a171d..0605f8f 100644
--- a/app/controllers/concerns/mission_control/jobs/job_filters.rb
+++ b/app/controllers/concerns/mission_control/jobs/job_filters.rb
@@ -9,7 +9,7 @@ module MissionControl::Jobs::JobFilters

   private
     def set_filters
-      @job_filters = { job_class_name: params.dig(:filter, :job_class_name).presence, queue_name: params.dig(:filter, :queue_name).presence }.compact
+      @job_filters = { job_class_name: params.dig(:filter, :job_class_name).presence, queue_name: params.dig(:filter, :queue_name).presence, finished_at: (params.dig(:finished_at_start)..params.dig(:finished_at_end)) }.compact
     end

     def active_filters?
diff --git a/lib/active_job/jobs_relation.rb b/lib/active_job/jobs_relation.rb
index 307b711..db698ff 100644
--- a/lib/active_job/jobs_relation.rb
+++ b/lib/active_job/jobs_relation.rb
@@ -25,7 +25,7 @@ class ActiveJob::JobsRelation
   STATUSES = %i[ pending failed in_progress blocked scheduled finished ]
   FILTERS = %i[ queue_name job_class_name ]

-  PROPERTIES = %i[ queue_name status offset_value limit_value job_class_name worker_id recurring_task_id ]
+  PROPERTIES = %i[ queue_name status offset_value limit_value job_class_name worker_id recurring_task_id finished_at ]
   attr_reader *PROPERTIES, :default_page_size

   delegate :last, :[], :reverse, to: :to_a
@@ -51,12 +51,13 @@ def initialize(queue_adapter: ActiveJob::Base.queue_adapter, default_page_size:
   # * <tt>:queue_name</tt> - To only include the jobs in the provided queue.
   # * <tt>:worker_id</tt> - To only include the jobs processed by the provided worker.
   # * <tt>:recurring_task_id</tt> - To only include the jobs corresponding to runs of a recurring task.
-  def where(job_class_name: nil, queue_name: nil, worker_id: nil, recurring_task_id: nil)
+  def where(job_class_name: nil, queue_name: nil, worker_id: nil, recurring_task_id: nil, finished_at: nil)
     # Remove nil arguments to avoid overriding parameters when concatenating +where+ clauses
     arguments = { job_class_name: job_class_name,
       queue_name: queue_name,
       worker_id: worker_id,
-      recurring_task_id: recurring_task_id
+      recurring_task_id: recurring_task_id,
+      finished_at: finished_at
     }.compact.collect { |key, value| [ key, value.to_s ] }.to_h

     clone_with **arguments
diff --git a/lib/active_job/queue_adapters/solid_queue_ext.rb b/lib/active_job/queue_adapters/solid_queue_ext.rb
index 9dc9865..2dbf944 100644
--- a/lib/active_job/queue_adapters/solid_queue_ext.rb
+++ b/lib/active_job/queue_adapters/solid_queue_ext.rb
@@ -173,7 +173,7 @@ def retry_all
         attr_reader :jobs_relation

         delegate :queue_name, :limit_value, :limit_value_provided?, :offset_value, :job_class_name,
-          :default_page_size, :worker_id, :recurring_task_id, to: :jobs_relation
+          :default_page_size, :worker_id, :recurring_task_id, :finished_at, to: :jobs_relation

         def executions
           execution_class_by_status
@@ -190,6 +190,7 @@ def finished_jobs
           SolidQueue::Job.finished
             .then { |jobs| filter_jobs_by_queue(jobs) }
             .then { |jobs| filter_jobs_by_class(jobs) }
+            .then { |jobs| filter_jobs_by_finish_time(jobs) }
             .then { |jobs| limit(jobs) }
             .then { |jobs| offset(jobs) }
         end
@@ -249,6 +250,10 @@ def filter_jobs_by_queue(jobs)
           queue_name.present? ? jobs.where(queue_name: queue_name) : jobs
         end

+        def filter_jobs_by_finish_time(jobs)
+          finished_at.present? ? jobs.where(finished_at: finished_at) : jobs
+        end
+
         def filter_executions_by_class(executions)
           job_class_name.present? ? executions.where(job: { class_name: job_class_name }) : executions
         end
```

Reply:
awesome, looks good :)

Reply:
manage to make it work for the range, had to change a couple of extra things, do you mind taking a look?